### PR TITLE
SOF-7341: add getMaterialFileConfig method

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

**Review by timurbazhirov — APPROVED** [review]

File: src/js/material.ts [hunk 1]
```diff
@@ -388,6 +405,37 @@ export function MaterialMixin<
...
+            const config: MaterialFileConfig = {
+                src: {
+                    extension: fileExtension,
+                    filename: fileName,
+                    text: fileContent,
+                    hash: CryptoJS.MD5(fileContent).toString(),
+                },
+            };
+            // add ICSD ID if present
+            if (fileExtension === "cif") {
```
Comment (timurbazhirov) [inline src/js/material.ts:426]: Let's simplify and remove the external and icsdId parts handling

File: src/js/material.ts [hunk 2]
```diff
@@ -388,6 +405,37 @@ export function MaterialMixin<
 
             return checks;
         }
+
+        /**
+         * @summary Returns a config to create a material from a CIF or POSCAR file.
+         */
+        static getMaterialFileConfig(
```
Comment (timurbazhirov) [inline src/js/material.ts:412]: constructMaterialSrcFromFile

File: src/js/material.ts (hunk 1 above)
Comment (k0stik) [reply src/js/material.ts]: Done

File: src/js/material.ts (hunk 2 above)
Comment (k0stik) [reply src/js/material.ts]: Done

**Pull request merged**